src/main.rs

```diff
@@ -277,12 +290,14 @@ fn handle_rate_update(
 fn handle_btc_balance_update(
     btc_balance_update: anyhow::Result<bitcoin::Amount>,
     maker: &mut Maker,
-    swarm: &mut libp2p::Swarm<Nectar>,
+    swarm: &mut Swarm,
 ) {
     match btc_balance_update {
         Ok(btc_balance) => match maker.update_bitcoin_balance(btc_balance) {
             Ok(Some(new_sell_order)) => {
-                swarm.orderbook.publish(new_sell_order.into());
+                if let Err(e) = swarm.make(new_sell_order.into()) {
+                    tracing::error!("Failed to make new order: {}", e)
+                };
```

Comment:
```suggestion
                let _ = swarm.make(new_sell_order.into()).map_err(|e| {
                    tracing::error!("Failed to make new order: {}", e)
                });
```

Do you really prefer the `if let Err(e) = function() { log(e) }` syntax? I find it very verbose as a reader. It focused the attention to the matching part `Err(e)` which is just about logging an error, taking it away from the actual function call `swarm.make` which is the important part of the line.

Reply:
Oh, yeah, I forgot that we could do it with `map_err`. That's what I was looking for.